File: internal/domain/booking.go
package domain

import "time"

type Booking struct {
	ID         string    `json:"id"`
	ServiceID  string    `json:"service_id"`
	StaffID    string    `json:"staff_id"`
	ClientID   string    `json:"client_id"`
	LocationID string    `json:"location_id"`
	StartAt    time.Time `json:"start_at"`
	EndAt      time.Time `json:"end_at"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

File: frontend/src/types/booking.ts
export interface Booking {
  id: string
  service_id: string
  service_name: string
  staff_id: string
  staff_name: string
  start_at: string
  end_at: string
  customer_name: string
  customer_email: string
  created_at: string
  updated_at: string
}

export interface CreateBookingRequest {
  service_id: string
  staff_id: string
  start_at: string // ISO date string
  customer_phone: string
  customer_name: string
  customer_email?: string
  location_id?: string
}